### DISCUSSION TITLE
Using UXF for eye-tracking experiment

### DISCUSSION THREAD

**Discussion by EfiLmp** [immersivecognition/unity-experiment-framework#154] [General]
Hello,

I am developing an experiment in unity and along with behavioral data I am aiming to collect pupil diameter data from HTC Vive Pro Eye while the participants are implementing the task. My experimental session works perfectly in computer mode (when only UXF package is imported in the project), however it is not executing in the VR environment, and no errors pop-up. 

To convert my project from computer to VR mode I have followed the guide for HTC Vive Pro Eye and imported all the packages as suggested: https://developer.tobii.com/xr/develop/unity/getting-started/vive-pro-eye/

Moreover, I have added the XRRig Prefab (from the XR Management Plugin) in the Hierarchy, to set up the VR Camera and the Controllers, and the SRanipal Eye Framework Prefab in order to get the eye tracking data.

Any guidance would be greatly appreciated! Thank you in advance for your time.

**Comment by CormacAltman** [comment]
Before reading note that I am not a programmer or associated with the project, so take my advice with a grain of salt. Not sure if you managed to figure out your issue with the VR environment as a whole, but there is a link to an eye tracking script the creator of UXF made for the vive here. It has pupil diameter as one of the tracked variables. 

My personal recommendation for getting an experiment to run in VR would be to remove the eye-tracking stuff and see if your project runs without it. If it does not, try downloading his tutorial project and seeing if that runs in VR. 

https://github.com/immersivecognition/unity-experiment-framework/wiki/Tracker-system

**Reply by EfiLmp** [reply]
Thank you so much for taking the time to reply. I managed to make my experiment work in VR!
As suggested I followed the tutorial project and the documentation, which helped me to make the appropriate adjustments on my project.

Thanks again!

**Comment by jackbrookes** [comment]
Hi, you can try adapting this code:

https://gist.github.com/jackbrookes/774af439d580cd9a9d3a8611b1167e5d

But @CormacAltman is right, you best route is to write a script like the above that inherits from `Tracker` and on each frame samples the relevant eye data. This will then be saved to a file per trial (if assigned under tracked objects in the session inspector)

**Reply by EfiLmp** [reply]
Dr. Brookes thank you for your reply!
Indeed I created a script that extracts and prints eye data per frame. However, I attached it in an empty GO, ending up with different file formats for my behavioural and eye tracking data. 
I will follow your suggestion and add them to the session inspector, seems very neat approach!

Thank you!
